keytree/src/lib.rs

```diff
@@ -18,6 +26,15 @@ impl Xprv {
         rng.fill_bytes(&mut dk);
         Xprv { scalar, dk }
     }
+
+    /// Returns a new Xpub, generated from the provided Xprv.
+    pub fn to_xpub(xprv: &Self) -> Xpub {
```

Comment:
use the keyword `self` as a designated first argument which turns the _function_ into a _method_:
```suggestion
    pub fn to_xpub(&self) -> Xpub {
```

Reply:
Note: the method can still be called as `Xprv::to_xpub(&xprv)`, but with use of `self` it can _also_ be called as a usual OOP-y method: `xprv.to_xpub()`.

Reply:
Thank you! I got it, now